server/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTests.java

```diff
@@ -93,16 +92,18 @@ private static void assertParseSearchSource(SearchSourceBuilder testBuilder, XCo
     }
 
     public void testSerialization() throws IOException {
-        SearchSourceBuilder testBuilder = createSearchSourceBuilder();
-        try (BytesStreamOutput output = new BytesStreamOutput()) {
-            testBuilder.writeTo(output);
-            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), namedWriteableRegistry)) {
-                SearchSourceBuilder deserializedBuilder = new SearchSourceBuilder(in);
-                assertEquals(deserializedBuilder, testBuilder);
-                assertEquals(deserializedBuilder.hashCode(), testBuilder.hashCode());
-                assertNotSame(deserializedBuilder, testBuilder);
-            }
-        }
+        SearchSourceBuilder original = createSearchSourceBuilder();
+        SearchSourceBuilder copy = copyBuilder(original);
+        assertEquals(copy, original);
+        assertEquals(copy.hashCode(), original.hashCode());
+        assertNotSame(copy, original);
+    }
+
+    public void testSerializingWithRuntimeFieldsBeforeSupportedThrows() {
+        SearchSourceBuilder original = new SearchSourceBuilder().runtimeMappings(randomRuntimeMappings());
+        Version v = Version.V_8_0_0.minimumCompatibilityVersion();
+        Exception e = expectThrows(IllegalArgumentException.class, () -> copyBuilder(original, v));
+        assertThat(e.getMessage(), equalTo("Versions before 8.0.0 don't support [runtime_mappings] and search was sent to [" + v + "]"));
```

Comment:
can you extract this behaviour to the description of the PR so it is highlighted? I think it makes sense but it differs from how we've done these things in the past.

Reply:
I do this most of the time I implement something. But, sure!